### PR TITLE
opentoonz: 1.5.0 -> 1.7.1

### PR DESCRIPTION
**Pull request by Naxdy**
## Description of changes

Update opentoonz to 1.7.1. Also move libtiff and opencv modifications into the `default.nix` as opposed to `all-packages.nix`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/graphics/opentoonz/default.nix [hunk 1]
```diff
@@ -38,6 +68,7 @@ in stdenv.mkDerivation rec {
   cmakeFlags = [
     "-DTIFF_INCLUDE_DIR=${libtiff.dev}/include"
     "-DTIFF_LIBRARY=${libtiff.out}/lib/libtiff.so"
+    "-DCMAKE_SKIP_BUILD_RPATH=ON"
```
Comment (AndersonTorres) [inline pkgs/applications/graphics/opentoonz/default.nix:71]:
```suggestion
    (lib.cmakeBool "CMAKE_SKIP_BUILD_RPATH" true)
```

File: pkgs/applications/graphics/opentoonz/source.nix [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+    opentoonz = "1.7.1";
+    libtiff = "4.0.3"; # The version in thirdparty/tiff-*
   };
 
   src = fetchFromGitHub {
     owner = "opentoonz";
     repo = "opentoonz";
     rev = "v${versions.opentoonz}";
-    sha256 = "1rw30ksw3zjph1cwxkfvqj0330v8wd4333gn0fdf3cln1w0549lk";
+    sha256 = "sha256-5iXOvh4QTv+G0fjEHU62u7QCee+jbvKhK0+fQXbdJis=";
```
Comment (AndersonTorres) [inline pkgs/applications/graphics/opentoonz/source.nix:14]:
```suggestion
    hash = "sha256-5iXOvh4QTv+G0fjEHU62u7QCee+jbvKhK0+fQXbdJis=";
```

**Comment by wegank** [comment]
Partially done in #257344, please rebase.

File: pkgs/applications/graphics/opentoonz/default.nix [hunk 3]
```diff
@@ -1,8 +1,38 @@
...
+, qtscript
+, qtserialport
+, lib
+, stdenv
+, superlu
+, wrapQtAppsHook
+, callPackage
+}:
+let
+  source = import ./source.nix { inherit fetchFromGitHub; };
```
[the remark this answers is not in the document]
Comment (Naxdy) [reply pkgs/applications/graphics/opentoonz/default.nix:28]: Makes sense to me, but probably best posed to @chkno

[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/applications/graphics/opentoonz/default.nix:28]: Theyn does not look active. That fork was not touched for years:

https://github.com/NixOS/nixpkgs/issues/290642

[the remark this answers is not in the document]
Comment (Naxdy) [reply pkgs/applications/graphics/opentoonz/default.nix]: Fair enough, went ahead and inlined `libtiff`!

**Pull request merged**